lib/node_modules/@stdlib/blas/ext/base/zfill/src/main.c

```diff
@@ -19,27 +19,41 @@
 #include "stdlib/blas/ext/base/zfill.h"
 #include "stdlib/complex/float64/ctor.h"
 #include "stdlib/strided/base/stride2offset.h"
-#include <stdint.h>
+#include "stdlib/blas/base/shared.h"
 
 /**
 * Fills a double-precision complex floating-point strided array with a specified scalar constant.
 *
-* @param N       number of indexed elements
-* @param alpha   scalar constant
-* @param X       input array
-* @param stride  index increment
+* @param N        number of indexed elements
+* @param alpha    scalar constant
+* @param X        input array
+* @param strideX  strided length
 */
-void c_zfill( const int64_t N, const stdlib_complex128_t alpha, stdlib_complex128_t *X, const int64_t stride ) {
-	int64_t ix;
-	int64_t m;
-	int64_t i;
+void API_SUFFIX(stdlib_strided_zfill)( const CBLAS_INT N, const stdlib_complex128_t alpha, stdlib_complex128_t *X, const CBLAS_INT strideX ) {
+	CBLAS_INT ox = stdlib_strided_stride2offset( N, strideX );
+	API_SUFFIX(stdlib_strided_zfill_ndarray)( N, alpha, X, strideX, ox );
+}
+
+/**
+* Fills a double-precision complex floating-point strided array with a specified scalar constant using alternative indexing semantics.
+*
+* @param N        number of indexed elements
+* @param alpha    scalar constant
+* @param X        input array
+* @param strideX  index increment
+* @param offsetX  starting index
+*/
+void API_SUFFIX(stdlib_strided_zfill_ndarray)( const CBLAS_INT N, const stdlib_complex128_t alpha, stdlib_complex128_t *X, const CBLAS_INT strideX, const CBLAS_INT offsetX ) {
+	CBLAS_INT ix;
+	CBLAS_INT m;
+	CBLAS_INT i;
 
 	if ( N <= 0 ) {
 		return;
 	}
 
```

Comment:
@headlessNode You have a bug in this implementation. Namely, instead of `i` in L62-L76, you need to be using `ix`. You are currently not accounting for the offset.

Reply:
I believe we should have unit tests which test whether the `ndarray` method supports providing "an `x` offset", similar to how we should have tests for whether the `ndarray` method supports "an `x` stride".

Reply:
doesn't the `the function supports an offset parameter` test case cover that?

Reply:
Yes, that was my point. Unit tests should have caught the native implementation bug.

Reply:
Could there be a bug in `isSameComplex128Array` that we are using to check equality?

Reply:
No. The issue is not `isSameComplex128Array`. The issue is that the following test conflates multiple behaviors:

```javascript
tape( 'the function supports an offset parameter', opts, function test( t ) {
	var expected;
	var x;

	x = new Complex128Array([
		1.0,
		-2.0,
		3.0,  // 0
		-4.0, // 0
		6.0,
		-8.0,
		10.0, // 1
		-12.0 // 1
	]);
	expected = new Complex128Array([
		1.0,
		-2.0,
		5.0,  // 0
		-5.0, // 0
		6.0,
		-8.0,
		5.0,  // 1
		-5.0  // 1
	]);

	zfill( 2, new Complex128( 5.0, -5.0 ), x, 2, 1 );
	t.strictEqual( isSameComplex128Array( x, expected ), true, 'returns expected value' );
	t.end();
});
```

Notice that this test tests both an offset and a non-unit stride. Hence, the code path for L42-L64 was not exercised, and thus the bug was not detected. I see, however, that this test originates from `dfill` and `sfill`. Better would have been to isolate the single behavior to test. E.g., 

```javascript
tape( 'the function supports an offset parameter', opts, function test( t ) {
	var expected;
	var x;

	x = new Complex128Array([
		1.0,
		-2.0,
		3.0,  // 0
		-4.0, // 0
		6.0,  // 1
		-8.0, // 1
		10.0,
		-12.0
	]);
	expected = new Complex128Array([
		1.0,
		-2.0,
		5.0,  // 0
		-5.0, // 0
		5.0,  // 1
		-5.0, // 1
		10.0,
		-12.0
	]);

	zfill( 2, new Complex128( 5.0, -5.0 ), x, 1, 1 );
	t.strictEqual( isSameComplex128Array( x, expected ), true, 'returns expected value' );
	t.end();
});
```

If you want to improve these tests similar to the above, feel free to submit a follow-up PR updating `dfill`, `sfill`, `cfill`, and `zfill`.

Reply:
Makes sense. I'll update the tests in follow-up.